src/utils/cachingAssets.ts

```diff
@@ -0,0 +1,43 @@
+import { ApolloClient, NormalizedCacheObject, gql } from '@apollo/client'
+
+type ReadUrlFromCacheArg = {
+  fileType: 'avatar' | 'cover'
+  channelId: string | null
+  client: ApolloClient<NormalizedCacheObject>
+}
+
+type WriteUrlInCacheArg = {
+  url: string | null
+} & ReadUrlFromCacheArg
+
+const cachedCoverUrlFragment = gql`
+  fragment CoverUrlField on Channel {
+    cachedCoverUrl
+  }
+`
+
+const cachedAvatarUrlFragment = gql`
+  fragment AvatarUrlField on Channel {
+    cachedAvatarUrl
+  }
+`
+
+export const writeUrlInCache: (arg: WriteUrlInCacheArg) => void = ({ url, fileType, channelId, client }) => {
```

Comment:
```suggestion
export const writeUrlInCache = ({ url, fileType, channelId, client }: WriteUrlInCacheArg) => {
```
I think we don't need to specify returned type here. Typescript will figure it out